wait/tls_test.go

```diff
@@ -0,0 +1,146 @@
+	certBytes []byte
+
+	//go:embed testdata/cert.key
+	keyBytes []byte
+)
+
+// testForTLSCert creates a new CertStrategy for testing.
+func testForTLSCert() *wait.TLSStrategy {
+	return wait.ForTLSCert(clientCertFilename, clientKeyFilename).
+		WithRootCAs(caFilename).
+		WithServerName(serverName).
+		WithStartupTimeout(time.Millisecond * 50).
+		WithPollInterval(time.Millisecond)
+}
+
+func TestForCert(t *testing.T) {
+	errNotFound := errdefs.NotFound(errors.New("file not found"))
+	ctx := context.Background()
+
+	t.Run("ca-not-found", func(t *testing.T) {
+		target := newRunningTarget()
+		target.EXPECT().CopyFileFromContainer(anyContext, caFilename).Return(nil, errNotFound)
+		err := testForTLSCert().WaitUntilReady(ctx, target)
+		require.EqualError(t, err, context.DeadlineExceeded.Error())
+	})
+
+	t.Run("cert-not-found", func(t *testing.T) {
+		target := newRunningTarget()
+		caFile := io.NopCloser(bytes.NewBuffer(caBytes))
+		target.EXPECT().CopyFileFromContainer(anyContext, caFilename).Return(caFile, nil)
+		target.EXPECT().CopyFileFromContainer(anyContext, clientCertFilename).Return(nil, errNotFound)
+		err := testForTLSCert().WaitUntilReady(ctx, target)
+		require.EqualError(t, err, context.DeadlineExceeded.Error())
+	})
+
+	t.Run("key-not-found", func(t *testing.T) {
+		target := newRunningTarget()
+		caFile := io.NopCloser(bytes.NewBuffer(caBytes))
+		certFile := io.NopCloser(bytes.NewBuffer(certBytes))
+		target.EXPECT().CopyFileFromContainer(anyContext, caFilename).Return(caFile, nil)
+		target.EXPECT().CopyFileFromContainer(anyContext, clientCertFilename).Return(certFile, nil)
+		target.EXPECT().CopyFileFromContainer(anyContext, clientKeyFilename).Return(nil, errNotFound)
+		err := testForTLSCert().WaitUntilReady(ctx, target)
+		require.EqualError(t, err, context.DeadlineExceeded.Error())
+	})
+
+	t.Run("valid", func(t *testing.T) {
+		target := newRunningTarget()
+		caFile := io.NopCloser(bytes.NewBuffer(caBytes))
+		certFile := io.NopCloser(bytes.NewBuffer(certBytes))
+		keyFile := io.NopCloser(bytes.NewBuffer(keyBytes))
+		target.EXPECT().CopyFileFromContainer(anyContext, caFilename).Return(caFile, nil)
+		target.EXPECT().CopyFileFromContainer(anyContext, clientCertFilename).Return(certFile, nil)
+		target.EXPECT().CopyFileFromContainer(anyContext, clientKeyFilename).Return(keyFile, nil)
+
+		certStrategy := testForTLSCert()
+		err := certStrategy.WaitUntilReady(ctx, target)
+		require.NoError(t, err)
+
+		pool := x509.NewCertPool()
+		require.True(t, pool.AppendCertsFromPEM(caBytes))
+		cert, err := tls.X509KeyPair(certBytes, keyBytes)
+		require.NoError(t, err)
+		got := certStrategy.TLSConfig()
+		require.Equal(t, serverName, got.ServerName)
+		require.Equal(t, []tls.Certificate{cert}, got.Certificates)
+		require.True(t, pool.Equal(got.RootCAs))
+	})
+}
+
+func ExampleForTLSCert() {
+	ctx := context.Background()
+
+	// waitForTLSCert {
+	forCert := wait.ForTLSCert("/app/tls.pem", "/app/tls-key.pem").
```

Comment:
suggestion: add a comment with the origin of the certificates, which are set in the Dockerfile. Maybe interesting to add a code block in the docs site showing the entire Dockerfile